### DISCUSSION TITLE
Prospective applicant for Kart CLI Help Improvements

### DISCUSSION THREAD

**Discussion by akkashyap8991** [koordinates/kart#580] [Google Soc / SoD]
Hi, 
I am Akhil N Kashyap and I am interested in working on the Kart CLI project. I am currently working as a Member of Technical Staff with Oracle, India. I graduated in 2020 with a B.Tech degree in Information Technology from the National Institute of Technology, Karnataka.
I am well versed in Python programming and Shell scripting, which seem to be relevant to this project.

I have prior experience in working on Command Line Interface tool using python in my current position . So, I feel that I am a good candidate to contribute to this particular project. 

I have briefly explored the code base and the CLI part of it, which primarily uses click.

It would be great if the mentors concerned can provide me a bit more details on the following :
- Is there a way to build the kart project locally and test the changes (especially the CLI part)
- What issues/bugs I can take up and start working on?
- Any other reference/link to help me understand the requirements better would be appreciated

Thanks,
Akhil

**Comment by rcoup** [comment]
Hi Akhil,

> Is there a way to build the kart project locally and test the changes (especially the CLI part)

Yes, see https://github.com/koordinates/kart/blob/master/CONTRIBUTING.md

> What issues/bugs I can take up and start working on?

We'll do some work over the next few days to label some issues with the "good first issue" label, maybe pick up one of them.

> Any other reference/link to help me understand the requirements better would be appreciated

We want to get an outcome along the lines of aws-cli or git's help systems, so they might be worth researching. Whatever solution we adopt needs an answer for each platform we support (macOS, Linux, Windows) since Windows in particular has a different way of working.

**Reply by akkashyap8991** [reply]
Thanks, will look at the aws-cli and git help and will try coming up with a suitable solution
